src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -191,6 +193,25 @@ void OTAImageProcessorImpl::HandleApply(intptr_t context)
     ESP_LOGI(TAG, "Applying, Boot partition set offset:0x%x", imageProcessor->mOTAUpdatePartition->address);
 
     // HandleApply is called after delayed action time seconds are elapsed, so it would be safe to schedule the restart
+    uint32_t softwareVersion;
+    DeviceLayer::ConfigurationMgr().GetSoftwareVersion(softwareVersion);
+    err = OTARequestorNVSHelper::WriteSoftwareVersion(softwareVersion);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Failed to write software version: (%s)", esp_err_to_name(err));
+        return;
+    }
+    NodeId nodeId;
+    FabricIndex fabIndex;
+    EndpointId endpointId;
+    OTARequestorInterface * requestorInstance = GetRequestorInstance();
+    requestorInstance->TestModeGetProviderParameters(nodeId, fabIndex, endpointId);
```

Comment:
We shouldn't be doing this (`TestModeGetProviderParameters` should not exist). The ProviderLocation should be persisted by the OTA core, and read from there. That currently does not exist. Perhaps a TODO to switch this over once that is in place?

Reply:
Done